packages/dds/tree/src/changeset/toDelta.ts

```diff
@@ -0,0 +1,362 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { unreachableCase } from "@fluidframework/common-utils";
+import { brand, fail } from "../util";
+import { FieldKey, Value } from "../tree";
+import { Delta, ProtoNode, Transposed as T } from ".";
+
+/**
+ * Converts a Changeset into a Delta.
+ * @param changeset - The Changeset to convert
+ * @returns A Delta for applying the changes described in the given Changeset.
+ */
+export function toDelta(changeset: T.Changeset): Delta.Root {
+    return convertPositionedMarks<Delta.OuterMark>(changeset.marks);
+}
+
+function convertPositionedMarks<TMarks>(marks: T.PositionedMarks): Delta.PositionedMarks<TMarks> {
+    const out: Delta.PositionedMarks<Delta.Mark> = [];
+    for (const offsetMark of marks) {
+        const offset = offsetMark.offset || 0;
+        const mark = offsetMark.mark;
+        if (Array.isArray(mark)) {
+            for (const attach of mark) {
+                // Inline into `switch(attach.type)` once we upgrade to TS 4.7
+                const type = attach.type;
+                switch (type) {
```

Comment:
```suggestion
                switch (attach.type) {
```

Reply:
As discussed, that breaks `unreachableCase(type)`.